core/trino-main/src/main/java/io/trino/execution/QueryPreparer.java

```diff
@@ -61,23 +64,42 @@ public PreparedQuery prepareQuery(Session session, Statement wrappedStatement)
     {
         Statement statement = wrappedStatement;
         Optional<String> prepareSql = Optional.empty();
-        if (statement instanceof Execute) {
-            prepareSql = Optional.of(session.getPreparedStatementFromExecute((Execute) statement));
+        if (statement instanceof Execute executeStatement) {
+            prepareSql = Optional.of(session.getPreparedStatementFromExecute(executeStatement));
             statement = sqlParser.createStatement(prepareSql.get(), createParsingOptions(session));
         }
-
-        if (statement instanceof ExplainAnalyze) {
-            Statement innerStatement = ((ExplainAnalyze) statement).getStatement();
+        else if (statement instanceof ExecuteImmediate executeImmediateStatement) {
+            // TODO: find a non-hacky way to get correct position information in case of syntactic and semantic errors,
+            //       perhaps handle this all within the analyzer by having dedicated logic to deal with the ExecuteImmediate
+            //       node there instead of doing it in this class
```

Comment:
I dealt with the same kind of issue when parsing JSON path within a query. Maybe you find it helpful: https://github.com/trinodb/trino/blob/71e2a87a6a4bef0df7ec9d717a669abe74e9d6fb/core/trino-parser/src/main/java/io/trino/sql/jsonpath/PathParser.java#L43